tools/src/main/java/org/apache/kafka/trogdor/coordinator/TaskManager.java

```diff
@@ -294,25 +298,82 @@ private JsonNode getCombinedStatus() {
     }
 
     /**
-     * Create a task.
-     *
-     * @param id                    The ID of the task to create.
-     * @param spec                  The specification of the task to create.
+     * Creates and schedules a task.
      */
-    public void createTask(final String id, TaskSpec spec)
+    public void createAndScheduleTask(TaskDetail task)
             throws Throwable {
+        ManagedTask managedTask = createTask(task);
+
+        long delayMs = scheduleTask(managedTask);
+        log.info("Created a new task {} with spec {}, scheduled to start {} ms from now.", task.id, task.spec, delayMs);
+    }
+
+    /**
+     * Atomically creates and schedules the given tasks. If any task fails creation, none get scheduled.
+     */
+    public void createAndScheduleTasksAtomic(List<TaskDetail> tasks) throws BadRequestException, RequestConflictException {
+        List<ManagedTask> managedTasks = new ArrayList<>();
+        if (tasks.stream().map(task -> task.id).collect(Collectors.toSet()).size() < tasks.size())
+            throw new RequestConflictException("Duplicate task IDs given");
+
```

Comment:
We should also check to make sure there are no invalid empty task IDs.

Reply:
That is handled in `TaskManager#createTask()`. In the worst case, we'd create N-1 tasks and throw an error on the last one, but none would get scheduled. Do you think it's worth to fail early?

Reply:
Like I wrote earlier, this should just be a map, so duplicates should not be a problem.  I think it would be good to do all the validation here.  There's no reason not to do it and it makes things more robust if the code is re-arranged in the future.